src/main/webapp/creatives/maps/simid_map_creative.js

```diff
@@ -0,0 +1,55 @@
+
+import BaseSimidCreative from '../base_simid_creative.js';
+
+/**
+ * A sample SIMID ad that shows a map of nearby locations
+ */
+
+const AdParamKeys = {
+  BUTTON_LABEL: 'buttonLabel',
+  SEARCH_QUERY: 'searchQuery',
+  MARKER: 'marker',
+};
+
+const FIND_NEAREST_TEMPLATE_TEXT = "Find Nearest ";
+const DEFAULT_BUTTON_LABEL = "Location";
+
+export default class SimidMapCreative extends BaseSimidCreative {
+
+  constructor() {
+    super();
+  }
+
+  /** @override */
+  onStart(eventData){
+    debugger;
+    super.onStart(eventData);
+    //ToDo(juliareichel@): handle invalid JSON and param errors
+    const adParams = JSON.parse(this.creativeData.adParameters);
+    const buttonLabel = adParams[AdParamKeys.BUTTON_LABEL]; 
+    const searchQuery = adParams[AdParamKeys.SEARCH_QUERY];
+    //ToDo(juliareichel@): handle case where searchQuery is undefined
+    const marker = adParams[AdParamKeys.MARKER];
+    this.specifyButtonFeatures_(buttonLabel);
+  }
+ 
+  /**
+   * Sets the text of the initial button and assigns it a click functionality.
+   * @param {string=} buttonLabel refers to the default value "location" written on the
+   * Find Nearest button. Advertisers have the option of changing the text on button.
```

Comment:
normally comments go

comment .......
@param {type} name description
@return {type} desc

So take this comment and move it up as it makes sense.

Reply:
The comment is the first thing? If you are referring to the Find Nearest button, that is part of the @param

Reply:
I got rid of the * in front of line 39 to make it more obvious